### DISCUSSION TITLE
Using multiple prisma schemas/runtimes exceeds lambda size limit

### DISCUSSION THREAD

**Discussion by morgs32** [prisma/prisma#16297] [Q&A]
We have multiple prisma "services" which means multiple dbs and multiple prisma schemas and we want to use them all (via SDKs) in one serverless function (usng TRPC). 

Is there a way for them to use one and the same ENGINE LIBRARY? Here you can see we have it 3 times which puts us over the max size for a lambda:

<img width="330" alt="image" src="https://user-images.githubusercontent.com/1364795/202172282-8d781c21-5155-486b-b7ba-58414c2fe57e.png">

I was thinking perhaps I could create a couple symlinks so they are pointing to the same file? Does anyone foresee that being a problem?

**Comment by nurul3101** [comment]
Hello @morgs32 👋 

Can you have a look at this [Blog](https://sabinadams.hashnode.dev/managing-lambda-size-with-multiple-prisma-clients)?
This covers the exact same scenario that you are facing and demonstrates how you can get around the lambda size limit when using multiple schemas. Also, please let us know if you have any feedback or run into any issues while following it, this is written by a Prisma Employee so we can update it to make sure it has all the information accurate and up to date.

**Comment by janpio** [comment]
This could help: https://www.prisma.io/docs/reference/api-reference/environment-variables-reference#prisma_query_engine_library

**Comment by morgs32** [comment]
@nurul3101 I'm having a very difficult time following that blog post with vercel. Here's an error in vercel:

<img width="1006" alt="image" src="https://user-images.githubusercontent.com/1364795/202308379-27c1900e-7481-4404-8452-2268d6b29f16.png">

I'll dig into what distinguishes the `var/task` directory from the `vercel/path0/` directory.

Perhaps I should dump the engine into the `/public` directory and then use the env var like @janpio  suggests?

**Reply by nurul3101** [reply]
Yes, can you please try the suggestion that Jan suggested?

**Reply by morgs32** [reply]
Unfortunately this ENV VAR seems ALSO to be used to copy an engine INTO the client directory.

<img width="1329" alt="image" src="https://user-images.githubusercontent.com/1364795/202441105-5ce9ed84-fa40-413a-98d8-9500f3a85656.png">

I'll try to delete the variable at build time... and see if it can still be used at runtime.

**Reply by vladi-stevanovic** [reply]
:wave: Hello @morgs32 , I’m going through pending GitHub Discussions to check if you need any help. Did deleting the variable at build time help?

Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! :raised_hands:

**Reply by morgs32** [reply]
Hi @vladi-stevanovic and whoever may be struggling with this. Here's an example project (on Github) of how you can achieve this: https://github.com/morgs32/multiple-prisma-schemas-on-vercel/tree/main

Another way to go about using multiple prisma dbs/schemas is to use Prisma proxy. It's doesn't require any of the somewhat ugly, definitely non-obvious workarounds from the implementation above. Plus, you can take advantage of vercel's edge functions if you want. 

But, for anyone NOT using prisma proxy but trying to isolate unrelated data into different dbs and schemas, this might help!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
